### PR TITLE
Render fancy ordered-list markers: honour list style/delimiter (#100)

### LINKED ISSUE #100: Render fancy ordered-list markers: honour list style/delimiter from QuantEcon/mystmd#50
QuantEcon/mystmd#50 parses Pandoc-style fancy ordered-list markers (`a.`, `iv.`, `(i)`) and the page AST served to the theme carries the metadata — `list` nodes gain `style` (`lower-alpha` | `upper-alpha` | `lower-roman` | `upper-roman`) and `delimiter` (`paren` | `parens`). But `myst-to-react`'s `list` handler only emits `<ol start id className>`, so HTML sites render plain decimal markers. See QuantEcon/mystmd#51 (canonical tracker) and QuantEcon/mystmd#74 (impact report: 10 enumerate labels silently became `1./2./3.` when the Deep Learning book upgraded qe-v6 → qe-v8).

This theme is the right place for a deployable fix: `app/root.tsx` already merges custom node renderers via `mergeRenderers([defaultRenderers, JUPYTER_RENDERERS])`, which is myst-theme's designed override point — no patch-package entry needed, and it ships to every book pinned to this template without waiting on upstream.

## Sketch

**1. Custom `list` renderer** merged into `RENDERERS`, replacing the default for ordered lists:

```tsx
const OL_TYPE: Record<string, string> = {
  'lower-alpha': 'a',
  'upper-alpha': 'A',
  'lower-roman': 'i',
  'upper-roman': 'I',
};

list({ node, className }) {
  if (!node.ordered) return /* default <ul> path */;
  return (
    <ol
      start={node.start || undefined}
      type={OL_TYPE[node.style]}
      id={node.html_id}
      className={classNames(className, node.delimiter && `delimiter-${node.delimiter}`)}
    >
      <MyST ast={node.children} />
    </ol>
  );
}
```

The `type` attribute alone makes browsers render `a.` / `i.` markers natively — that covers the period-delimited forms with zero CSS.

**2. CSS in `styles/`** for the paren forms, using the built-in `list-item` counter keyed off the type attribute (safer than `::marker { content: ... }`, which Safari only recently supports):

```css
ol.delimiter-parens, ol.delimiter-paren { list-style: none; }
ol.delimiter-parens > li::before { content: '(' counter(list-item) ') '; }
ol.delimiter-paren  > li::before { content: counter(list-item) ') '; }
ol[type='a'].delimiter-parens > li::before { content: '(' counter(list-item, lower-alpha) ') '; }
ol[type='A'].delimiter-parens > li::before { content: '(' counter(list-item, upper-alpha) ') '; }
ol[type='i'].delimiter-parens > li::before { content: '(' counter(list-item, lower-roman) ') '; }
ol[type='I'].delimiter-parens > li::before { content: '(' counter(list-item, upper-roman) ') '; }
/* plus the same four for single-paren, and indent/positioning for the ::before marker */
```

Verify `counter(list-item)` picks up the `start` attribute in all target browsers (it should — it's the spec'd list-item counter — but nested-list and `start` edge cases are worth a fixture page).

## Test case

```markdown
(a) Alpha item body.

(i) Roman item body.
```

should render markers `(a)` / `(i)`, not `1.` / `1.` — plus period forms `a.` / `iv.` via the `type` attribute.

## Relationship to upstream

This fixes books pinned to this template. Stock `book-theme` sites need the same change in jupyter-book/myst-theme's `myst-to-react`, which can only ship paired with upstreaming fancy-lists itself (upstream's parser doesn't emit `style`/`delimiter`) — that pairing is planned in the `fancy-lists` entry of QuantEcon/mystmd's `quantecon/UPSTREAM-PRS.yml`. When it lands upstream, this override becomes redundant and can be dropped.

### PR DESCRIPTION
**Pull request by mmcky**
Closes #100.

## What

Fancy ordered lists (QuantEcon/mystmd#50) — markers like `a.`, `iv.`, `(i)` — parse into `list` nodes carrying `style` and `delimiter`, but `myst-to-react`'s `list` renderer drops both fields, so HTML sites render plain decimal markers. This landed as a silent regression for paren-form documents when books upgraded past the fancy-lists release: the literal `(a)` text disappears and a `1.` takes its place (QuantEcon/mystmd#74, found upgrading the Deep Learning book).

This PR overrides the `list` renderer through the theme's existing `mergeRenderers` extension point in `app/root.tsx` — no patch-package entry, no myst-theme fork:

- **`app/renderers.tsx`** — `LIST_RENDERERS` maps `style` to the `<ol type>` attribute (`lower-alpha` → `a`, `upper-roman` → `I`, …) and exposes `delimiter` as a `delimiter-paren` / `delimiter-parens` class. Plain ordered lists and unordered lists emit byte-identical HTML to the default renderer.
- **`styles/lists.css`** — restates the `type` mapping as author CSS, because the `type` attribute is only a presentational hint and loses to any author `list-style-type` (prose typography sets `decimal`); draws `(a)` / `a)` markers with the built-in `list-item` CSS counter (honours `<ol start>`), positioned in the list gutter so wrapped lines align like native markers.

## Verification

SSR-rendered list ASTs through the exact merged renderer stack from `root.tsx` (`defaultRenderers` + `JUPYTER_RENDERERS` + `LIST_RENDERERS`):

| input | emitted |
| --- | --- |
| plain decimal | `<ol start="1">` (unchanged) |
| `(a)` lower-alpha parens | `<ol start="1" type="a" class="delimiter-parens">` |
| `(i)` lower-roman parens | `<ol start="1" type="i" class="delimiter-parens">` |
| `B)` upper-alpha paren, start 2 | `<ol start="2" type="A" class="delimiter-paren">` |
| `iv.` period style | `<ol start="4" type="i">` (no class) |
| unordered | `<ul>` (unchanged) |

`npm run compile` (tsc) passes; `npm run build:css` passes and the emitted bundle contains the new rules.

## Test drive

A minimal page reproduces QuantEcon/mystmd#74:

```markdown
(a) Alpha item body.

(i) Roman item body.
```

Markers should read `(a)` / `(i)` — decimals before this change. The PR preview workflow builds real lectures, though note the preview corpus may not contain fancy-list markers, in which case the preview mainly confirms no regression to ordinary lists.

## Lifetime

Interim override: the long-term home is jupyter-book/myst-theme's `myst-to-react`, which can only be proposed once fancy-lists itself goes upstream to jupyter-book/mystmd (upstream's parser doesn't emit `style`/`delimiter`). That sequencing is tracked in QuantEcon/mystmd#51 — when it lands and this theme picks up the fixed `myst-to-react`, drop `app/renderers.tsx` and `styles/lists.css`.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: app/renderers.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  list({ node, className }: { node: GenericNode; className?: string }) {
+    if (node.ordered) {
+      const delimiterClass =
+        node.delimiter && node.delimiter !== 'period' ? `delimiter-${node.delimiter}` : undefined;
+      return (
+        <ol
+          start={node.start || undefined}
+          type={node.style ? OL_TYPE[node.style] : undefined}
+          id={node.html_id}
+          className={[className, delimiterClass].filter(Boolean).join(' ') || undefined}
```
[the remark this answers is not in the document]
Comment (mmcky) [reply app/renderers.tsx]: Fixed in 4fb2c13 — both lookups are now typed allow-lists (`OL_TYPE` keyed by `OlStyle`, and a `DELIMITER_CLASS` table replacing the `!== 'period'` test, so unknown values emit nothing rather than an arbitrary `delimiter-*` class), and `start` uses `?? undefined` so CommonMark's `0.` start survives. SSR-verified: `start: 0` now emits `start="0"`, and a hostile `delimiter: "bogus injected"` emits no class.

File: app/renderers.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  'upper-alpha': 'A',
+  'lower-roman': 'i',
+  'upper-roman': 'I',
+};
+
+export const LIST_RENDERERS: NodeRenderers = {
+  list({ node, className }: { node: GenericNode; className?: string }) {
+    if (node.ordered) {
+      const delimiterClass =
+        node.delimiter && node.delimiter !== 'period' ? `delimiter-${node.delimiter}` : undefined;
```
[the remark this answers is not in the document]
Comment (mmcky) [reply app/renderers.tsx]: Added in 4fb2c13 — with one twist: the harness builds the fixture with stock npm mystmd, which doesn't parse fancy-list markers (the parser is QuantEcon/mystmd#50), so a literal `(a)` in a fixture page would have parsed as plain paragraph text and exercised nothing. Instead the new `lists` page writes ordinary `1.` lists and a fixture-local transform (`fancy-lists.mjs`) stamps `style`/`delimiter` onto them, keeping coverage independent of the building CLI. This coverage immediately caught a real bug: HTML matches `[type=...]` case-insensitively, so the `ol[type='a']`/`ol[type='A']` CSS selected the same elements and every lowercase variant rendered uppercase — the CSS now keys off case-sensitive `list-*` classes. Also added a `lists-markers` DOM assertion test, since a wrong marker case moves too few pixels for the snapshot's 1% diff budget. Darwin baselines committed; Linux ones to follow via /update-new-snapshots.
